cmd/stackdriver-prometheus-sidecar/main.go

```diff
@@ -258,20 +263,35 @@ func main() {
 	level.Info(logger).Log("host_details", Uname())
 	level.Info(logger).Log("fd_limits", FdLimits())
 
-	promExporter, err := oc_prometheus.NewExporter(oc_prometheus.Options{
-		Registry: prometheus.DefaultRegisterer.(*prometheus.Registry),
-	})
-	if err != nil {
-		level.Error(logger).Log("msg", "Creating Prometheus exporter failed", "err", err)
-		os.Exit(1)
-	}
-	view.RegisterExporter(promExporter)
-
 	httpClient := &http.Client{Transport: &ochttp.Transport{}}
 
 	if *projectId == "" {
 		*projectId = getGCEProjectID()
 	}
+
+	for _, backend := range cfg.monitoringBackends {
+		if backend == "prometheus" {
+			promExporter, err := oc_prometheus.NewExporter(oc_prometheus.Options{
+				Registry: prometheus.DefaultRegisterer.(*prometheus.Registry),
+			})
+			if err != nil {
+				level.Error(logger).Log("msg", "Creating Prometheus exporter failed", "err", err)
+				os.Exit(1)
+			}
+			view.RegisterExporter(promExporter)
+		}
+		if backend == "stackdriver" {
+			sd, err := oc_stackdriver.NewExporter(oc_stackdriver.Options{ProjectID: *projectId})
+			if err != nil {
+				level.Error(logger).Log("msg", "Creating Stackdriver exporter failed", "err", err)
+				os.Exit(1)
+			}
+			defer sd.Flush()
+			view.RegisterExporter(sd)
+			view.SetReportingPeriod(60 * time.Second)
+		}
```

Comment:
Minor, `else { level.Warning(logger).Log("msg", "Unknown backend", "backend", backend) }` (this might be easier to write as a switch/case)

Reply:
Enums-based flags already provide validation of values, but `switch` statement here seems more appropriate anyway, so I've made the change.